connectors/llm/interface.py

```diff
@@ -38,28 +43,55 @@ def ask(self, system_prompt, previous_messages, question, agent_id, stream):
                 if "title" in metadata:
                     title = metadata["title"]
                     context_text += f", document title: '{title}'"
-                context_text += ">>\n\n" f"{page_content}\n\n" f"<<Search result {i+1} END>>\n"
+                context_text += (">>\n\n" f"{page_content}\n\n" f"<<Search result {i+1} END>>\n")
 
         prompt = ChatPromptTemplate.from_template(cfg.USER_PROMPT_TEMPLATE)
         prompt_params = {"context": context_text, "question": question}
```

Comment:
Here's where we become aware of the question content. We need to somehow count the number of tokens for the question somewhere around here and add it to the running total right?